feat: Arnold licensing error for mtoa5.4.7.1

ZainAallii opened the pull request:
Fixes: *NA*

### What was the problem/requirement? (What/Why)
A customer reported an issue where their Maya job succeeds even when abort_on_license_fail option is enabled and license checkout for Arnold fails. This is happening because the string which we are trying to match in adaptor.py line#217 is different from what is actually printing in the log. What was the solution? (How)
I changed the string to catch both the error messages:
- aborting render because the abort_on_license_fail option was enabled
- borting render because this is a batch render and abort_on_license_fail option is enabled

### What is the impact of this change?
It will catch Arnold license error an fail the job.

### How was this change tested?
I ran the unit test by running `hatch run all:test`, I got `Required test coverage of 41.0% reached. Total coverage: 44.05%`

- Have you run the unit tests?
Yes. The result is here: 
[MayaUnitTest.txt](https://github.com/user-attachments/files/18714996/MayaUnitTest.txt)


### Did you run the "Job Bundle Output Tests"? If not, why not? If so, paste the test results here.

No, I ran the adaptor locally and tested it with real scene file. I did not have any licenses for Arnold so my job failed on the error message. 
![MayaAdaptor](https://github.com/user-attachments/assets/876a3bc1-4dfa-4f5c-a664-0a74b537a0ea)
Check the logs: 
[MayaAdaptorLocalRun.txt](https://github.com/user-attachments/files/18714999/MayaAdaptorLocalRun.txt)

### Was this change documented?
NA

### Is this a breaking change?
No
----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

ZainAallii committed:
feat: Arnold licensing error for mtoa5.4.7.1

Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

jairaws committed:
fix: OutputPath is read from the scene in the submitter (#231)

* fix: Load scene settings when the submitter is open

Signed-off-by: Jair Ruiz <167156521+jairaws@users.noreply.github.com>

* fix: OutputPath is read from the scene in the submitter

Signed-off-by: Jair Ruiz <167156521+jairaws@users.noreply.github.com>

---------

Signed-off-by: Jair Ruiz <167156521+jairaws@users.noreply.github.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

jairaws committed:
fix: Update the warning message used when the scene has unsaved changes (#238)

Signed-off-by: Jair Ruiz <167156521+jairaws@users.noreply.github.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

Ahuge committed:
fix: Use `byFrameStep` instead of `byFrame` (#224)

This is talked about in the following link:
https://help.autodesk.com/view/MAYAUL/2023/ENU/?guid=__Nodes_renderGlobals_html

Essentially `byFrame` is no longer used but kept for backwards
compatibility. And the value it gives us is different than
`byFrameStep`

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

jairaws commented on test/unit/deadline_adaptor_for_maya/MayaAdaptor/test_adaptor.py:
```diff
@@ -866,7 +866,9 @@ def test_error_on_arnold_license_fail(
         adaptor = MayaAdaptor(init_data)
         expected_regex_list = [
-            re.compile("(aborting render because the abort_on_license_fail option was enabled)")
+            re.compile(
```
This unit test only checks that the expected regexes are in the list of callback regexes. Do we have a unit test to check that the error message is caught for all cases? If so, we should update that unit test to include the new case. If not, we should add a unit test to validate that the error messages are caught.

ZainAallii replied:
I did test it on with Maya Arnold job (MTOA5.4.7.1) manually. Please check the logs attached https://github.com/user-attachments/files/18714999/MayaAdaptorLocalRun.txt

jairaws replied:
Synced up with the requester offline. There's some reluctance to add the unit test based on time concerns. However, I believe we should promote high quality standards for our code. Thus, IMO, we should add test coverage for every behaviour change, even if there wasn't a test for similar behaviour before.

I wrote the test myself to help unblock this PR:
```
from typing import List
...
    @pytest.mark.parametrize(
        "message, should_match",
        [
            ("aborting render because the abort_on_license_fail option was enabled", True),
            (
                "aborting render because this is a batch render and abort_on_license_fail option is enabled",
                True,
            ),
            ("aborting render because ", False),
            ("aborting render because something else happened", False),
        ],
    )
    def test_regex_for_error_on_arnold_license_fail(
        self, message: str, should_match: bool
    ) -> None:
        """
        Tests that the regexes used to catch arnold license errors work as expected.
        """
        # GIVEN
        regex_list = [
            re.compile(
                "(aborting render because (?:the abort_on_license_fail option was enabled|this is a batch render and abort_on_license_fail option is enabled))"
            )
        ]

        # WHEN
        results: List[bool] = [regex.match(message) for regex in regex_list]

        # THEN
        assert any(results) == should_match
```

@ZainAallii please add this test and update the PR. Feel free to modify it or add any other test case.

ZainAallii replied:
Hello @jairaws 
I added the test as suggested. 
[test.txt](https://github.com/user-attachments/files/19171892/test.txt)

jairaws replied:
Thank you!

skarpree committed:
fix: Tighten error regex pattern.

Added error regex to the adaptor for capturing render failure for Maya. I have picked some Deadline 10 regex handler for fatal error Maya reports.

Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

skarpree committed:
fix: Changed the cameras dropdown wording option from "All Cameras" to "All Renderable Cameras" in the submitter

fixed the wording in the submitter from "All Cameras" to "All Renderable Cameras" in the Cameras dropdown under "Job-specific settings"

Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

jericht committed:
chore(test): broaden setuptools-scm regex to fix test (#243)

Signed-off-by: Jericho Tolentino <68654047+jericht@users.noreply.github.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

skarpree committed:
Revert "Merge branch 'skarpree-submitter-cameras' of https://github.com/skarpree/deadline-cloud-for-maya into skarpree-submitter-cameras"

This reverts commit 86a0523f3a412b9e31397e119360224f5d46ec72, reversing
changes made to 33238c6e2aef17a9ad6a100661b6bae815ef33e7.

Signed-off-by: Karpreet Sethi <skarpree@amazon.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

joel-wong-aws committed:
chore: add maintenance label to maintenance GitHub template (#241)

Signed-off-by: Joel Wong <127782171+joel-wong-aws@users.noreply.github.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

sakshie95 committed:
chore: pinned version of deadline to 0.49.6 for now until next release (#247)

Signed-off-by: Sakshi <sakshie@amazon.com>
Co-authored-by: Sakshi <sakshie@amazon.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

jericht committed:
chore(release): switch to python-semantic-release 9.12.* (#249)

Signed-off-by: Jericho Tolentino <68654047+jericht@users.noreply.github.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

client-software-ci committed:
chore(release): 0.15.2 (#250)

* chore(release): 0.15.2

Signed-off-by: client-software-ci <129794699+client-software-ci@users.noreply.github.com>

* chore: updated changelog

Signed-off-by: Sakshi <sakshie@amazon.com>

---------

Signed-off-by: client-software-ci <129794699+client-software-ci@users.noreply.github.com>
Signed-off-by: Sakshi <sakshie@amazon.com>
Co-authored-by: Sakshi <sakshie@amazon.com>
Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

ZainAallii committed:
feat:Arnold test for mtoa5.4.7.1 error

Signed-off-by: Zain Ali <55154081+ZainAallii@users.noreply.github.com>

jairaws commented on .github/ISSUE_TEMPLATE/maintenance.yml:
```diff
@@ -1,7 +1,7 @@
 description: Some type of improvement
 title: "Maintenance: (short description of the issue)"
-labels: ["needs triage"]
+labels: ["maintenance", "needs triage"]
```
Something seems off with you branch. Some of these changes are from other PRs. Did you `pull --rebase` your commit before pushing it?
